modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/TableKeyDetails/index.js

```diff
@@ -0,0 +1,300 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayIcon from '@clayui/icon';
+import classNames from 'classnames';
+import getCurrentEndDate from '../../../../common/utils/getCurrentEndDate';
+import {
+	getFormatedProductName,
+	getProductDescription,
+	getProductName,
+	getStatusActivationTag,
+	hasVirtualCluster,
+} from '../../containers/DXPActivationKeysTable/utils/constants';
+
+const HOST_NAME = 'Host Name';
+const IP_ADDRESSES = 'IP Addresses';
+const MAC_ADDRESSES = 'Mac Addresses';
+const SUBSCRIPTION_IMAGE_FILE = 'dxp_icon.svg';
+
+const NO_EXPIRATION_DATE = 100;
+
+const TableKeyDetails = ({
+	activationKeys,
+	assetsPath,
+	setValueToCopyToClipboard,
+}) => {
+	const instanceSizeFormated = activationKeys.sizing.slice(7, 8);
+
+	const unlimitedLicenseDate = new Date().setFullYear(
+		new Date().getFullYear() + NO_EXPIRATION_DATE
+	);
+
+	const handleExpiredDate =
+		new Date(activationKeys.expirationDate) >=
+		new Date(unlimitedLicenseDate)
+			? 'Does Not Expire'
+			: getCurrentEndDate(activationKeys.expirationDate);
+
+	const handleCopyToClipboard = (value, action) => {
+		navigator.clipboard.writeText(action);
+
+		setValueToCopyToClipboard(value);
+	};
```

Comment:
When we use the browser API should be good use inside an useEffect.
```suggestion
	useEffect(() => {
	  if (action) {
	    navigator.clipboard.writeText(action);
	  }
	}, [action]);
```

You can try to transform the action in a state value and check inside the useEffect to use the browse API.